### PR TITLE
Add Baidu Provider

### PR DESCRIPTION
**Pull request by ArcherTrister**

### CODE REVIEW

File: src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /// Contains constants specific to the <see cref="BaiduAuthenticationHandler"/>.
+    /// </summary>
+    public static class BaiduAuthenticationConstants
+    {
+        public static class Claims
+        {
+            /// <summary>
+            /// 当前登录用户的数字ID
+            /// </summary>
+            public const string UserId = "urn:baidu:userid";
```
Comment (kinosang) [inline src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs:19]: Use `ClaimTypes.NameIdentifier`

File: src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            /// <summary>
+            /// 当前登录用户的用户名，值可能为空。
+            /// </summary>
+            public const string UserName = "urn:baidu:username";
```
Comment (kinosang) [inline src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs:23]: Use `ClaimTypes.Name`

File: src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            /// </summary>
+            public const string Portrait = "urn:baidu:portrait";
+            /// <summary>
+            /// 自我简介，可能为空。
+            /// </summary>
+            public const string UserDetail = "urn:baidu:userdetail";
+            /// <summary>
+            /// 生日，以yyyy-mm-dd格式显示。
+            /// </summary>
+            public const string Birthday = "urn:baidu:birthday";
```
Comment (kinosang) [inline src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationConstants.cs:39]: Use `ClaimTypes.DateOfBirth`

**Comment by martincostello** [comment]
Is this a duplicate of #245?

**Comment by kinosang** [comment]
@martincostello yep

**Comment by martincostello** [comment]
Closing as this is a duplicate of #245.

**Pull request closed without merging**